src/sentry/tasks/data_export.py

```diff
@@ -44,20 +49,26 @@ def assemble_download(data_export_id):
                     file = File.objects.create(
                         name=file_name, type="export.csv", headers={"Content-Type": "text/csv"}
                     )
-                    file.putfile(tf)
+                    file.putfile(tf, logger=logger)
                     data_export.finalize_upload(file=file)
+                    logger.info("dataexport.end", extra={"id": data_export_id})
             except IntegrityError as error:
-                capture_exception(error)
+                metrics.incr("dataexport.error", instance=error)
+                logger.error(
+                    six.text_type("dataexport.error: {}").format(error),
+                    extra={"query": data_export.payload, "org": data_export.organization_id},
+                )
                 raise DataExportError("Failed to save the assembled file")
     except DataExportError as error:
-        # TODO(Leander): Implement logging
         return data_export.email_failure(message=error)
     except NotImplementedError as error:
-        # TODO(Leander): Implement logging
         return data_export.email_failure(message=error)
     except BaseException as error:
-        # TODO(Leander): Implement logging
-        capture_exception(error)
+        metrics.incr("dataexport.error", instance=error)
```

Comment:
do we want to be passing the whole error as the instance, like would the error message be enough? And if so, it seems like we should use `six.text_type(error)`

Reply:
fixed up in e5a4a38e6e2d0ad4ef768c588d5b358e3fab2007 !